### PR TITLE
Added ESLint Rule to GOCD Plugin

### PR DESCRIPTION
**Pull request by grvpandey11**
## Hey, I just made a Pull Request!

Added an optional ESLint rule - no-top-level-material-ui-4-imports -in good plugin which has an auto fix function to migrate the imports and used it to migrate the Material UI imports for plugins/gocd.

Issue for reference: https://github.com/backstage/backstage/issues/23467

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: .changeset/unlucky-clouds-wash.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-gocd': patch
+---
+
+Added an optional ESLint rule - no-top-level-material-ui-4-imports -in gocd plugin which has an auto fix function to migrate the imports and used it to migrate the Material UI imports for plugins/gocd.
```
Comment (awanlin) [inline .changeset/unlucky-clouds-wash.md:5]:
```suggestion
Added an optional ESLint rule - no-top-level-material-ui-4-imports - which has an auto fix function to migrate the imports and used it to migrate the Material UI imports.
```

File: plugins/gocd/.eslintrc.js
Comment (awanlin) [inline plugins/gocd/.eslintrc.js:1]: Please make sure the file has a newline at the end

**Review by awanlin — CHANGES_REQUESTED** [review]
Thanks @grvpandey11 just a few nits

**Review by awanlin — APPROVED** [review]
✅ looks good, thanks @grvpandey11

**Review by freben — APPROVED** [review]

**Pull request merged**